ql/src/java/org/apache/hadoop/hive/ql/udf/UDFSubstr.java

```diff
@@ -174,8 +174,10 @@ public StatEstimator getStatEstimator() {
     }
 
     private Optional<Double> getRangeWidth(Range range) {
-      if (range.minValue != null && range.maxValue != null) {
-        return Optional.of(range.maxValue.doubleValue() - range.minValue.doubleValue());
+      if (range != null) {
```

Comment:
could you please also add the testcase from the jira?

Reply:
Thanks for your suggestion. Added test case substr_estimate_range.q.